### PR TITLE
Extends screenshot API

### LINKED ISSUE #3761: Extend screenshots API
**Is your feature request related to a problem? Please describe.**
The API for screenshots is still missing some bits:

* Delete screenshots, DELETE /api/screenshots/(int: id)/
* Remove unit association from screenshot, DELETE /api/screenshots/(int: id)/units/(int: unitid)/
* Edit screenshot, PUT/PATCH /api/screenshots/(int: id)/

**Describe the solution you'd like**
Complete CRUD for screenshots.

**Additional context**
Followup for https://github.com/WeblateOrg/weblate/issues/3076

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/91349010-extend-screenshots-api?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SaptakS**
Closes #3761 

- [x] Every commit has a message which describes it
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them
- [x] Any code changes come with test case
- [x] Any new functionality is covered by user documentation

### CODE REVIEW

File: weblate/api/views.py [hunk 1]
```diff
@@ -1154,6 +1154,24 @@ def units(self, request, **kwargs):
...
+
+        try:
+            source_string = obj.component.source_translation.unit_set.get(
+                pk=int(unit_id)
+            )
+        except (Unit.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```
Comment (nijel) [inline weblate/api/views.py:1171]: Can you please cover this by unit tests?

**Comment by nijel** [comment]
@SaptakS Seems like you didn't notice my feedback here ;-).

File: weblate/api/views.py (hunk 1 above)
Comment (SaptakS) [reply weblate/api/views.py:1171]: Added.

**Comment by SaptakS** [comment]
> @SaptakS Seems like you didn't notice my feedback here ;-).

Sorry, somehow it got overlooked. Made the changes as requested. Apologies for the delay.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**